py/server/deephaven/table.py

```diff
@@ -55,13 +60,282 @@
+    and types from the source table of the RollupTable. """
+
+
+class _FormatOperationsRecorder(Protocol):
+    """A mixin for creating format operations to be applied to individual nodes of either RollupTable or TreeTable."""
+
+    def format_column(self, formulas: Union[str, List[str]]):
+        """Returns a new recorder with the :meth:`~deephaven.table.Table.format_columns` operation applied to nodes."""
+        formulas = to_sequence(formulas)
+        j_format_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JFormatOperationsRecorder)
+        return self.__class__(j_format_ops_recorder.formatColumns(formulas))
+
+    def format_row_where(self, cond: str, formula: str):
+        """Returns a new recorder with the :meth:`~deephaven.table.Table.format_row_where` operation applied to
+        nodes."""
+        j_format_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JFormatOperationsRecorder)
+        return self.__class__(j_format_ops_recorder.formatRowWhere(cond, formula))
+
+    def format_column_where(self, col: str, cond: str, formula: str):
+        """Returns a new recorder with the :meth:`~deephaven.table.Table.format_column_where` operation applied to
+        nodes."""
+        j_format_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JFormatOperationsRecorder)
+        return self.__class__(j_format_ops_recorder.formatColumnWhere(col, cond, formula))
+
+
+class _SortOperationsRecorder(Protocol):
+    """A mixin for creating sort operations to be applied to individual nodes of either RollupTable or
+    TreeTable."""
+
+    def sort(self, order_by: Union[str, Sequence[str]]):
+        """Returns a new recorder with the :meth:`~deephaven.table.Table.sort` operation applied to nodes."""
+        order_by = to_sequence(order_by)
+        j_sort_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JSortOperationsRecorder)
+        return self.__class__(j_sort_ops_recorder.sort(order_by))
+
+    def sort_descending(self, order_by: Union[str, Sequence[str]]):
+        """Returns a new recorder with the :meth:`~deephaven.table.Table.sort_descending` applied to nodes."""
+        order_by = to_sequence(order_by)
+        j_sort_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JSortOperationsRecorder)
+        return self.__class__(j_sort_ops_recorder.sortDescending(order_by))
+
+
+class _FilterOperationsRecorder(Protocol):
+    """A mixin for creating filter operations to be applied to individual nodes of either RollupTable or
+    TreeTable."""
+
+    def where(self, filters: Union[str, Filter, Sequence[str], Sequence[Filter]]):
+        """Returns a new recorder with the :meth:`~deephaven.table.Table.where` operation applied to nodes."""
+        filters = to_sequence(filters)
+        j_filter_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JFilterOperationsRecorder)
+        return self.__class__(j_filter_ops_recorder.where(filters))
+
+
+class RollupNodeOperationsRecorder(JObjectWrapper, _FormatOperationsRecorder,
```

Comment:
Confirm that the private mixin classes display properly in Sphinx.

Reply:
These private classes aren't included in the generated docs but the methods all show up in the subclasses, which is exactly what we want to see.